# Natural hybridisation reduces vulnerability to climate change 

This repository contains code to perform the analyses and modeling described in the following paper:

Brauer CJ, Sandoval-Castillo J, Gates K, Hammer M, Unmack PJ, Bernatchez L, and Beheregaray, LB "Natural hybridisation reduces vulnerability to climate change (*accepted*) Nature Climate Change.


![Figure 1](../main/images/Figure1_hires_vector.jpg) "**Figure 1. Sampling locations and spatial patterns of hybridisation for *Melanotaenia splendida*, Malanda rainbowfish, *M. eachamensis*, *M. utcheensis* and Tully rainbowfish.** A) Sampling sites in the Wet Tropics of Queensland, Australia. B) Admixture plots for K=5 ancestral species. C) Treemix maximum likelihood tree showing introgression among species. D) Topographic relief profile indicating the difference in elevation between upland and lowland habitat along a transect between sites 13 and 35."


## Abstract

Under climate change, species unable to track their niche via range shifts are largely reliant on genetic variation to adapt and persist. Genomic vulnerability predictions are used to identify populations that lack the necessary variation, particularly at climate relevant genes. However, hybridization as a source of novel adaptive variation is typically ignored in genomic vulnerability studies. We estimated environmental niche models and genomic vulnerability for closely related species of rainbowfish (*Melanotaenia* spp.) across an elevational gradient in the Australian wet tropics. Hybrid populations between a widespread generalist and several narrow range endemic species exhibited reduced vulnerability to projected climates compared to pure narrow endemics. Overlaps between introgressed and adaptive genomic regions were consistent with a signal of adaptive introgression. Our findings highlight the often-underappreciated conservation value of hybrid populations and indicate that adaptive introgression may contribute to evolutionary rescue of species with narrow environmental ranges.

## Code

1. ENMs.R – Code to replicate the environmental niche models.
2. GEA.R – Code to perform the genotype–environments association redundancy analyses.
3. GV.R – Code to run the genomic vulnerability analyses

## Datasets

The *Melanotaenia duboulayi* reference genome assembly is available at NCBI GenBank accession number JAPDEC000000000.

[SNP genotypes (Figshare)](https://doi.org/10.6084/m9.figshare.21692918)


## A zip file containing all data and code for the analyses can be downloaded [here](https://github.com/pygmyperch/NER/archive/refs/heads/main.zip)